### DISCUSSION TITLE
Optional pull request promote/deploy

### DISCUSSION THREAD

**Discussion by cIvanrc** [community/community#12894] [Actions] [answered]
Hi

I find myself in a situation where I would like to be able to deploy a pull request, but not in a mandatory way for the workflow, since not all pull requests have an interest in deploying it.
Since before merging a pull request I would like to be able to deploy to the test/staging environment, but this is not strictly necessary for each and every one of the pull requests.
Currently there is something fascinating that is being able to approve deployments by environments.

I would like something like the developers could request manually if they want their pull request to be deployed to the test environment and this request can be approved or not (as it currently works), but that it be optional, that it does not stop the pull request to be able to merge it without having gone through a deployment, do not remove its "green" status

Does this make sense? Is there someone in the same situation? This has already been resolved and I just don't know such a solution?

**Comment by ethomson** [comment, marked as answer]
Interesting question @clvanrc - you might also try asking in the [GitHub Community Forum](https://github.community/c/code-to-cloud/52) which is has more Q&A. (This feedback section is more about giving feedback.) I think that you might find some people who have solved similar problems or created similar workflows.

I appreciate the feedback here, though, it helps us guide product direction. Thanks much.
